### PR TITLE
adds PDFSinglePageViewer.js file and example

### LINKED ISSUE #8188: Missing references in internal links in pageviewer example
Hi,

I'm trying different components of pdfjs to build my own viewer. My idea is based on the "pageviewer" component example because I want to load only a page at a time. Using this example is working ok, but I don't get the internal links to other pages (although I only load one each time).

I'm using the following pdf (in page 3) as an example: http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/pdf_open_parameters.pdf

I have added two statements to try to mimick what the simpleviewer example is doing with the linkservice, but it's not adding the page reference in the links. 
```
var pdfLinkService = new PDFJS.PDFLinkService(); // <--- ADDED
pdfLinkService.setDocument(pdfDocument, null); // <--- ADDED
```
(next is the modified version with these two lines)

My question is, is there a configuration that allows the pageviewer example generate links with page references?

**Modified versión of "pageviewer" example:**

```
'use strict';

if (!PDFJS.PDFViewer || !PDFJS.getDocument) {
  alert('Please build the pdfjs-dist library using\n' +
        '  `gulp dist`');
}

// The workerSrc property shall be specified.
//
PDFJS.workerSrc = '../../build/dist/build/pdf.worker.js';

var DEFAULT_URL = '../../web/pdf_open_parameters.pdf';
var PAGE_TO_VIEW = 3;
var SCALE = 1.0;

var container = document.getElementById('pageContainer');
var pdfLinkService = new PDFJS.PDFLinkService(); // <--- ADDED

// Loading document.
PDFJS.getDocument(DEFAULT_URL).then(function (pdfDocument) {

  pdfLinkService.setDocument(pdfDocument, null); // <--- ADDED

  // Document loaded, retrieving the page.
  return pdfDocument.getPage(PAGE_TO_VIEW).then(function (pdfPage) {
    // Creating the page view with default parameters.
    var pdfPageView = new PDFJS.PDFPageView({
      container: container,
      id: PAGE_TO_VIEW,
      scale: SCALE,
      defaultViewport: pdfPage.getViewport(SCALE),
      // We can enable text/annotations layers, if needed
      textLayerFactory: new PDFJS.DefaultTextLayerFactory(),
      annotationLayerFactory: new PDFJS.DefaultAnnotationLayerFactory()
    });
    // Associates the actual page with the view, and drawing it
    pdfPageView.setPdfPage(pdfPage);
    return pdfPageView.draw();
  });
});
```
----------------------

Link to PDF file (or attach file here):
http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/pdf_open_parameters.pdf

Configuration:
- Web browser and its version: Chrome, 56.0.2924.87
- Operating system and its version: Windows 10
- PDF.js version: 
{
  "version": "1.7.329",
  "build": 329,
  "commit": "25f772a"
}
- Is an extension:
No

Steps to reproduce the problem:
1. Open the provided pdf file and go to page 3 that have links to other pages. Links to other pages have the page reference missing, only shows a hash.
2.

What is the expected behavior? (add screenshot)
I was expecting to have at least the page parameter in the href attribute, something like "#page=x"

What went wrong? (add screenshot)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by nish17**
Please leave a comment in case you find anything wrong with this patch. I'll try my best to figure the issue and try to fix it.

Fixes #8188.

### CODE REVIEW

File: examples/components/pdfSinglePageView.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+if (!PDFJS.PDFSPViewer || !PDFJS.getDocument) {
+  alert('Please build the pdfjs-dist library using\n' +
+        '  `gulp dist-install`');
```
Comment (jsoref) [inline examples/components/pdfSinglePageView.js]: Out of curiosity: why the leading space on the new line?

File: web/PDFSinglePageViewer.js [hunk 2]
```diff
@@ -0,0 +1,944 @@
...
+    if (!this.pdfDocument) {
+      return;
+    }
+    this._setScale(val, false);
+  }
+
+  /**
+   * @returns {number}
+   */
+  get pagesRotation() {
```
Comment (jsoref) [inline web/PDFSinglePageViewer.js]: Unless you have a good reason, this concept should be page rotation not pages rotation

Comment (nish17) [inline web/PDFSinglePageViewer.js]: Actually, even in the main repository, that name is used. Do you want me to change the name from "pagesRotation" to "page_rotation"?

Comment (jsoref) [inline web/PDFSinglePageViewer.js]: Want is a strong word. I'm not very associated with the project.

And I definitely don't know about still wrt underscore/camel case

Comment (nish17) [inline web/PDFSinglePageViewer.js]: Okay then keeping it as it is.

**Comment by timvandermeij** [comment]
Closing since this is now implemented in #8724. We still need an example for that though, which is tracked in #8951. Feel free to work on that, which should be easy now that the single page viewer already exists.

**Pull request closed without merging**